R/lint.R

```diff
@@ -446,23 +454,27 @@ check_ranges <- function(ranges, max_col) {
     return()
   }
   if (!is.list(ranges)) {
-    stop("`ranges` must be NULL or a list.", call. = FALSE)
+    cli_abort(c(
+      i = "{.arg ranges} must be {.code NULL} or a {.cls list}.",
+      x = "Instead, it was {.cls {class(ranges)}}."
+    ))
   }
 
   for (range in ranges) {
     if (!is_number(range, 2L)) {
-      stop("`ranges` must only contain length 2 integer vectors without NAs.", call. = FALSE)
+      cli_abort("{.arg ranges} must only contain {.cls integer} vectors of length 2 without {.code NA}s.")
     } else if (!is_valid_range(range, max_col)) {
-      stop(sprintf(
-        "All entries in `ranges` must satisfy 0 <= range[1L] <= range[2L] <= nchar(line) + 1 (%d).", max_col
-      ), call. = FALSE)
+      cli_abort(c(
+        x = "Invalid range specified.",
+        i = "Argument {.arg ranges} must satisfy 0 <= range[1L] <= range[2L] <= nchar(line) + 1 ({.val {max_col}})."
```

Comment:
aside: in #2551 we specifically stopped `range[1L]==0`, but here it's valid. Does that mean we solved that issue incorrectly? Maybe it's something that should be handled in the SARIF code specifically instead? cc @AshesITR

Reply:
IINM Jump-to-source in RStudio accepts 0 as start-of-line while 1 jumps to after the first character. I see two options: Disallow 0 in general or handle it in sarif output.

If I indeed recall correctly about the special behavior for column 0, handling this case in sarif output seems acceptable. Do you have the time to test this?

Reply:
It's more that I don't really use RStudio these days, not sure how to test it (or at least, don't want to spend that much time setting up/figuring out testing).

I don't see anything about 0/1 mentioned in the `insertText()` documentation, though "start of document" is referred to as `c(1,1)`:

https://rstudio.github.io/rstudioapi/reference/rstudio-documents.html

Reply:
Forgot to link this back: https://github.com/rstudio/rstudioapi/issues/299

Reply:
I don't consider this to be a blocker for this PR. If something needs to be changed about the message here, that's a separate issue than using `{cli}` for formatting messages.